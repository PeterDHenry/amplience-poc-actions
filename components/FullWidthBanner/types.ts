export type FullWidthBannerProps = {
  _meta: {
    name: string
    schema: string
    deliveryId: string
  }
  image: { name: string }
  textAndCTA: {
    callToActionURL: string
    callToActionText: string
    headline: string
    strapline: string
  }
}

export const defaultProps: FullWidthBannerProps = {
  _meta: {
    name: 'string',
    schema: 'string',
    deliveryId: '1543512973',
  },
  image: { name: 'homepageh1d' },
  textAndCTA: {
    callToActionURL: 'https://www.wiggle.co.uk/run-new-season-2021',
    callToActionText: 'Default cta text',
    headline: 'Default header',
    strapline: 'Default subheader',
  },
}
