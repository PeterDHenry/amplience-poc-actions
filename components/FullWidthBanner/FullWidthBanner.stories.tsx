import React from 'react'
import { Story, Meta } from '@storybook/react'
import FullWidthBanner from './FullWidthBanner'
import { FullWidthBannerProps } from './types'

export default {
	title: 'CMS Content/FullWidthBanner',
	component: FullWidthBanner
} as Meta


export const fullWidthBanner: Story<FullWidthBannerProps> = (props)=><FullWidthBanner {...props} />
