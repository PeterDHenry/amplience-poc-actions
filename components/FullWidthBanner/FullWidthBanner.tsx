import React from 'react'
import styled from 'styled-components'
import { FullWidthBannerProps, defaultProps } from './types'

const ComponentContainer = styled.div`
  position: relative;
  max-height: 350px;
  overflow: hidden;
`

const Link = styled.a`
  text-decoration: none;
  text-shadow: 1px 1px #fff;
  color: #222;
  &:hover {
    color: #222;
  }
`
const Img = styled.img`
  max-width: 100%;
`

const ContentContainer = styled.div`
  width: 100%;
  position: absolute;
  bottom: 10%;
  left: 50%;
  transform: translate(-50%, -10%);
  text-align: center;
`

const H2 = styled.h2`
  text-transform: uppercase;
  font-size: 56px;
  font-weight: bold;
  margin-bottom: 30px;
`

const H3 = styled.h3`
  text-transform: capitalize;
  font-size: 38px;
  margin-bottom: 35px;
`

const Button = styled.button`
  padding: 5px 15px;
  text-transform: uppercase;
  color: #222;
  border: 2px solid #222;
  background: #fff;

  &:hover {
    background: #222;
    color: #ffd067;
  }
`

export function FullWidthBanner({
  image: { name },
  textAndCTA: { callToActionURL, callToActionText, headline, strapline },
}: FullWidthBannerProps) {
  const imgUrl = `https://cdn.media.amplience.net/i/wigglecrcnonprd/${name}`
  return (
    <div className="container pt-5">
      <div className="row">
        <ComponentContainer className="col-xs-12 col-sm-12">
          {
            <Link href={callToActionURL}>
              <Img src={imgUrl} alt="" />
              <ContentContainer>
                <H2> {headline} </H2>
                <H3> {strapline} </H3>
                <Button> {callToActionText} </Button>
              </ContentContainer>
            </Link>
          }
        </ComponentContainer>
      </div>
    </div>
  )
}

FullWidthBanner.defaultProps = defaultProps
