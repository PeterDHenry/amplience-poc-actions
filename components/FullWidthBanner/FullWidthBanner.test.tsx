import { render } from '@testing-library/react'
import FullWidthBanner, { BannerProps } from './FullWidthBanner'
import '@testing-library/jest-dom'

describe('Full width banner component', () => {
  test('Matches snapshot', () => {
    const { asFragment } = render(<FullWidthBanner {...BannerProps} />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  test('Should render FullWidthBanner component', () => {
    const { container } = render(<FullWidthBanner {...BannerProps} />)
    expect(container.getElementsByTagName('div')).toBeTruthy()
  })

  test('FullWidthBanner to have image with property "src" and end point link', () => {
    render(<FullWidthBanner {...BannerProps} />)
    const imgEl = document.querySelector('img')
    const imgSrc = imgEl?.getAttribute('src')
    const linkEl = document.querySelector('a')
    const linkElHref = linkEl?.getAttribute('href')

    expect(imgEl).toHaveAttribute('src')
    expect(imgSrc).toBeTruthy()
    expect(linkEl).toHaveAttribute('href')
    expect(linkElHref).toBeTruthy()
  })
})
