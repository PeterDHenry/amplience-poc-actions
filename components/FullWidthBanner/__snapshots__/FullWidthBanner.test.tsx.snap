// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Full width banner component Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container pt-5"
  >
    <div
      class="row"
    >
      <div
        class="FullWidthBanner__ComponentContainer-sc-zya7j3-0 dvlunK col-xs-12 col-sm-12"
      >
        <a
          class="FullWidthBanner__Link-sc-zya7j3-1 ffeCnq"
          href="https://www.wiggle.co.uk/run-new-season-2021"
        >
          <img
            alt=""
            class="FullWidthBanner__Img-sc-zya7j3-2 hcNdEH"
            src="https://cdn.media.amplience.net/i/wigglecrcnonprd/homepageh1d"
          />
          <div
            class="FullWidthBanner__ContentContainer-sc-zya7j3-3 djfYCM"
          >
            <h2
              class="FullWidthBanner__H2-sc-zya7j3-4 iYgKlo"
            >
               Default header 
            </h2>
            <h3
              class="FullWidthBanner__H3-sc-zya7j3-5 jSdkrl"
            >
               Default subheader 
            </h3>
            <button
              class="FullWidthBanner__Button-sc-zya7j3-6 eGNlge"
            >
               Default cta text 
            </button>
          </div>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
